sklearn/compose/tests/test_column_transformer.py

```diff
@@ -250,6 +254,84 @@ def transform(self, X, y=None):
     assert_array_equal(ct.transformers_[-1][2], [1])
 
 
+def test_column_transformer_dataframe_int_index():
```

Comment:
please add a short comment describing the test
(same for the other one)

Have you considered using parametrization for the cases and for the use of the callable? Not sure how worth it'd be